packages/wdio-config/src/lib/ConfigParser.js

```diff
@@ -85,21 +86,29 @@ export default class ConfigParser {
         /**
          * run single spec file only, regardless of multiple-spec specification
          */
-        if (typeof object.spec === 'string') {
-            const specs = []
-            const specList = object.spec.split(/,/g)
+        if (spec.length > 0) {
+            const specs = new Set()
+            const allSpecs = ConfigParser.getFilePaths(this._config.specs)
 
-            for (let spec of specList) {
-                if (fs.existsSync(spec)) {
-                    specs.push(path.resolve(process.cwd(), spec))
+            spec.forEach((spec_file) => {
+                if (fs.existsSync(spec_file)) {
+                    specs.add(path.resolve(process.cwd(), spec_file))
                 }
-            }
+                else {
+                    // Check for any specs that match a patter provided
+                    allSpecs.forEach((file) => {
+                        if (file.match(spec_file)) {
```

Comment:
Would this mean that if the applied `--spec` parameter is not within the `specs` option file list it will be excluded? Shouldn't we rather allow it like: run all my specs __and__ this file.

Reply:
Yes, that is currently what v4 is doing. I pulled that code over from there and refactored it a bit. I'm not sure if someone using the ```spec``` arg would ever want all of the tests to be ran. 

We use the pattern matching option a lot where I work. Lots of times we don't know the exact path to the file without looking it up and gets tedious typing it out so we are able to do something like this:

```./node_modules/.bin/wdio wdio.conf.js --spec foobar```

which then matches any file that contains ```foobar``` in it as long as the files fall inside of the ```specs``` option in the config.

Reply:
Thats the v4 version right now.

https://github.com/webdriverio/webdriverio/blob/master/lib/utils/ConfigParser.js#L166

Reply:
ok, looks good to me then